### ISSUE TITLE
Blurry interface using layershell

### ISSUE THREAD

**Issue by MalpenZibo** [waycrate/exwlshelleventloop#71]
Text on the interface seems blurry compared to the one obtained using "standard" iced or the pop_os fork (to access Wayland layershell).

Could my HiDPI monitor cause it (Resolution: 2880x1800)? 
I'm using Hyprland with a 2.0 scaling)

![screenshot-20241003-10:54:07](https://github.com/user-attachments/assets/6a8fb0ba-c0f0-42d8-9e90-b193d4d969e9)

**Comment by Decodetalkers** [comment]
I have not handle the case for scale.. I do not know how to handle it will be better

**Comment by Decodetalkers** [comment]
I have made a pr https://github.com/waycrate/exwlshelleventloop/pull/73 , but there is something I cannot solved.. the size of viewport.. I do not quite familiar with iced_graphic.. I need some help

**Comment by MalpenZibo** [comment]
I will try my best to help you!

**Comment by MalpenZibo** [comment]
The blurry effect is gone but now the application renders double the size. There should be somewhere an option to say to the iced layout system to limit the size of the main container.

**Comment by Decodetalkers** [comment]
> The blurry effect is gone but now the application renders double the size. There should be somewhere an option to say to the iced layout system to limit the size of the main container.

This is what I do not know.. I do not know much about iced.. I try do what iced_winit do, but the size of viewport is wrong,, I do not know what happened, so I need help

**Comment by MalpenZibo** [comment]
I'm looking here `https://github.com/MalpenZibo/iced_sctk` this is an """hard fork""" of the work done by the pop_os team on their fork of iced. I'm currently using this code to run my status bar. I have just removed the pop_os customization on iced maintaining only the support for the layershell protocol to use iced 12. 

I notice that they set up the surface with the physical size (for example in my configuration a width of 1440 px creates a surface with a width of 2880 px) and when they create the Viewport they use a physical size and a scale (in my case 2880 px and 2.0 of scaling) but then they call
```rust
self.viewport = Viewport::with_physical_size(physical_size, scale);
if let Some(wp_viewport) = self.wp_viewport.as_ref() {
  wp_viewport.set_destination(
    logical_size.width.ceil() as i32,
    logical_size.height.ceil() as i32,
  );
}
```

I'm not an expert but the wp_viewport is a `WpViewport` that seems a Wayland protocol. This is what I found reading the Rust wayland-protocol library documentation
```
pub mod viewporter {
    //! Viewporter protocol
    //!
    //! Provides the capability of scaling and cropping surfaces, decorrelating the surface
    //! dimensions from the size of the buffer.

    wayland_protocol!("./protocols/stable/viewporter/viewporter.xml", []);
}
```

So it seems more complicated than I expected.

**Comment by Decodetalkers** [comment]
@MalpenZibo in that pr I try to use that protocal, the scale seems be right, but still some problem with the scale. Please give me more help

**Comment by MalpenZibo** [comment]
I will try to implement the scaling. I will not have so much time the next week but I will try

**Comment by Decodetalkers** [comment]
finished, it is a mistake by myself before.

**Comment by Decodetalkers** [comment]
I have publish 0.9.0-beta1, you can take a try. I solve the two problems now

**Comment by MalpenZibo** [comment]
Well, the blurry effect seems better but the interface is strange. Everything seems too big, so for example in the counter_multi example the central button is collapsed because there isn't enough space.

I will try to make some changes to address this issue

**Comment by MalpenZibo** [comment]
@Decodetalkers 
Here https://github.com/MalpenZibo/exwlshelleventloop/pull/1/files

This is a POC that works very well on my machine (I used the main branch of this repository before merging #73).

I used fixed values (2 as scaling factor for example) that should be dynamic, also the `iced_sessionlock` doesn't compile and maybe there's a better way to handle the `wp_viewport` reference. 

Also, I'm not sure how to handle the scaling if the `Viewporter protocol` is not available (maybe the pop_os fork of iced has a fallback/legacy system to handle that case).

Anyway it looks very promising if you want to take a look, I will continue the work in the future but this week I don't have time :sweat_smile:
